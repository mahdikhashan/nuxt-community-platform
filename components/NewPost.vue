<template>
  <button
    class="flex gap-4 bg-green-500 hover:bg-green-600 px-6 text-white justify-center items-center py-3 rounded-full shadow-xl"
  >
    <svg class="w-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" aria-hidden="true">
      <title>New Post</title>
      <path
        d="M8,0A1,1,0,0,1,9,1V7h6a1,1,0,0,1,.993.883L16,8a1,1,0,0,1-1,1H9v6a1,1,0,0,1-.883.993L8,16a1,1,0,0,1-1-1V9H1a1,1,0,0,1-.993-.883L0,8A1,1,0,0,1,1,7H7V1A1,1,0,0,1,7.883.006Z"
        fill="currentColor"
      />
    </svg>
    <pre class="font-semibold">New Post</pre>
  </button>
</template>

<script>
export default {
  name: 'Button',
  props: {
  }
}
</script>