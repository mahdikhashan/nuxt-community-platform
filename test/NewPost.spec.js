import { mount } from '@vue/test-utils'
import NewPost from '@/components/NewPost.vue'

describe('NewPost', () => {
  const wrapper = mount(NewPost)

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('accepts click events', async () => {
    await wrapper.find('button').trigger('click')
  })
})
