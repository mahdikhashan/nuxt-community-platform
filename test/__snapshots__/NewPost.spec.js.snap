// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPost is a Vue instance 1`] = `
"<button class=\\"
    flex
    gap-4
    bg-green-500
    hover:bg-green-600
    px-6
    text-white
    justify-center
    items-center
    py-3
    rounded-full
    shadow-xl
  \\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"w-4\\">
    <title>New Post</title>
    <path d=\\"M8,0A1,1,0,0,1,9,1V7h6a1,1,0,0,1,.993.883L16,8a1,1,0,0,1-1,1H9v6a1,1,0,0,1-.883.993L8,16a1,1,0,0,1-1-1V9H1a1,1,0,0,1-.993-.883L0,8A1,1,0,0,1,1,7H7V1A1,1,0,0,1,7.883.006Z\\" fill=\\"currentColor\\"></path>
  </svg> <pre class=\\"font-semibold\\">New Post</pre></button>"
`;
